src/ngraph/op/not.cpp

```diff
@@ -26,9 +26,14 @@ op::Not::Not(const shared_ptr<Node>& arg)
     constructor_validate_and_infer_types();
 }
 
+// TODO(amprocte): Update this to allow only boolean, for consistency with logical binops.
```

Comment:
Discovered while writing unit tests for this PR that we are being too permissive here. Will open a separate PR to address that, since it is potentially a breaking change.